src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -751,6 +756,117 @@ public Task SuggestIndexMerge()
 
         }
 
+        [RavenAction("/databases/*/indexes/try", "POST", AuthorizationStatus.ValidUser)]
+        public async Task TestJavaScriptIndex()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var input = await context.ReadForMemoryAsync(RequestBodyStream(), "TestJavaScriptIndex");
+                if (input.TryGet("Definition", out BlittableJsonReaderObject index) == false)
+                    ThrowRequiredPropertyNameInRequest("Definition");
+
+                var indexDefinition = JsonDeserializationServer.IndexDefinition(index);
+
+                if (indexDefinition.Maps == null || indexDefinition.Maps.Count == 0)
+                    throw new ArgumentException("Index must have a 'Maps' fields");
+
+                indexDefinition.Type = indexDefinition.DetectStaticIndexType();
+
+                if (indexDefinition.Type.IsJavaScript() == false)
+                    throw new UnauthorizedAccessException("Testing indexes is only allowed for JavaScript indexes.");
+
+                if (indexDefinition.Name.StartsWith(Constants.Documents.Indexing.SideBySideIndexNamePrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new ArgumentException(
+                        $"Index name must not start with '{Constants.Documents.Indexing.SideBySideIndexNamePrefix}'. Provided index name: '{indexDefinition.Name}'");
+                }
+
+                var compiledIndex = new JavaScriptIndex(indexDefinition, Database.Configuration);
+
+                var pageSize = GetIntValueQueryString("pageSize", false) ?? defaultPageSizeForTestingJavaScriptIndex;
+                var collections = new HashSet<string>(compiledIndex.Maps.Keys);
+                var docsPerCollection = new Dictionary<string, List<DynamicBlittableJson>>();
+                using (context.OpenReadTransaction())
+                {
+                    foreach (var collection in collections)
+                    {
+                        docsPerCollection.Add(collection,
+                            Database.DocumentsStorage.GetDocumentsFrom(context, collection, 0, 0, pageSize).Select(d => new DynamicBlittableJson(d)).ToList());
+                    }
+
+                    var mapRes = new List<ObjectInstance>();
+                    //all maps
+                    foreach (var ListOfFunctions in compiledIndex.Maps)
+                    {
+                        //multi maps per collection
+                        foreach (var mapFunc in ListOfFunctions.Value)
+                        {
+                            foreach (var res in mapFunc(docsPerCollection[ListOfFunctions.Key]))
```

Comment:
How would this handle a multi map index on different collections?

Reply:
I don't follow your question, this is the code that is doing exactly what you're asking about.

Reply:
docsPerCollection is input per collection, this is why an array of ids is going to be annoying since i'll have to read them check their metadata bucket them into their respected collection...